x/bank/legacy/v043/store.go

```diff
@@ -14,8 +14,8 @@ import (
 // ref: https://github.com/cosmos/cosmos-sdk/issues/7092
 func migrateSupply(store sdk.KVStore, cdc codec.BinaryMarshaler) error {
 	// Old supply was stored as a single blob under the SupplyKey.
-	var oldSupply types.Supply // nolint:staticcheck
-	err := cdc.UnmarshalBinaryBare(store.Get(v040bank.SupplyKey), &oldSupply)
+	var oldSupply types.SupplyI // nolint:staticcheck
+	err := cdc.UnmarshalInterface(store.Get(v040bank.SupplyKey), &oldSupply)
```

Comment:
Unfortunately, what we're saving to state is an Any with the `types.Supply` struct 🤦